src/components/PopupWithForm.js

```diff
@@ -24,11 +25,16 @@ export default class PopupWithForm extends Popup {
     this._popupForm.addEventListener("submit", this._onPopupSubmit);
   }
 
-  _onPopupSubmit = (evt) => {
-    const event = evt;
-    this._handleFormSubmit(this._getInputValues(), event);
+  closeAfterSubmit() {
     this._popupForm.reset();
+    this._submitButton.textContent = this._submitButtonText;
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

Reply:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

So, if you call `closeAfterSubmit` in `finally`, then you need to delete `closing` and resetting, `toggleButtonState` from it (they should be in `then`)